semantic-conventions/src/opentelemetry/semconv/main.py

```diff
@@ -258,9 +307,30 @@ def setup_parser():
     subparsers = parser.add_subparsers(dest="flavor")
     add_code_parser(subparsers)
     add_md_parser(subparsers)
+    add_compat_check_parser(subparsers)
 
     return parser
 
 
+def download_previous_version(version: str) -> str:
+    filename = f"v{version}.zip"
+    tmppath = tempfile.mkdtemp()
+    path_to_zip = os.path.join(tmppath, filename)
+    path_to_semconv = os.path.join(tmppath, f"v{version}")
+
+    semconv_vprev = (
+        f"https://github.com/open-telemetry/semantic-conventions/archive/{filename}"
+    )
+
+    request = requests.get(semconv_vprev, allow_redirects=True, timeout=30)
```

Comment:
shouldn't we check 200? if unknown version is provided, 404 with html instead of zip file will be returned

Reply:
good catch!